wgpu-hal/src/vulkan/instance.rs

```diff
@@ -750,6 +750,17 @@ impl crate::Instance<super::Api> for super::Instance {
                 create_info = create_info.push_next(vk_create_info);
             }
 
+            let gpu_assisted_validation = &[vk::ValidationFeatureEnableEXT::GPU_ASSISTED];
+            let mut gpu_assisted_validation = vk::ValidationFeaturesEXT::builder()
+                .enabled_validation_features(gpu_assisted_validation)
+                .build();
+            if desc
+                .flags
+                .intersects(wgt::InstanceFlags::GPU_BASED_VALIDATION)
+            {
+                create_info = create_info.push_next(&mut gpu_assisted_validation);
+            }
```

Comment:
I don't think we can add this without first verifying that the validation layers are there - vulkan requires that we only ask for what is available.

Reply:
I'm not sure how to do this yet, since this is my first time working with vulcan, but I'll give it a crack!

Reply:
I _think_ I handled this with 17c854c3c.

Reply:
[`VK_EXT_validation_features`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_EXT_validation_features.html) is the required extension to use this functionality.

It's an instance extension so we need to ask for it [here](https://github.com/gfx-rs/wgpu/blob/b7b7f7daed5defc85a8bc64b3cf16057c2175f59/wgpu-hal/src/vulkan/instance.rs#L227) and then check if the instance actually supports it. If it doesn't, we can't use the functionality.

It's worth noting that the extension is deprecated and was subsumed into [`VK_EXT_layer_settings`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_EXT_layer_settings.html).